### PR TITLE
chore: switch to alpine based image

### PR DESCRIPTION
**Pull request by ksurl**
An alpine image based reduces the image size by almost half without further optimizations.

image

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # syntax=docker/dockerfile:1
-FROM node:16
+FROM node:16-alpine
 WORKDIR /automatisch
 
 RUN apt-get update && apt-get install -y postgresql-client
```
Comment (barinali) [inline docker/Dockerfile:5]: I think we need to update this branch with the `main` as apt-get does not exist for alpine. It currently fails when I try it.

**Review by barinali — CHANGES_REQUESTED** [review]

**Comment by ksurl** [comment]
What does running dos2unix on the file do? I believe it starts with LF line endings, and running docker on windows is still running a linux vm underneath, so it should not have formatting problems? Wouldn't it be better to handle it outside the image build?

if we need to do it in the image, I can break it into a multistage build so we don't have to worry about package cleanup. apt doesn't exist on alpine, the package manager is apk.

**Comment by barinali** [comment]
> What does running dos2unix on the file do? I believe it starts with LF line endings, and running docker on windows is still running a linux vm underneath, so it should not have formatting problems? Wouldn't it be better to handle it outside the image build?
> 
> if we need to do it in the image, I can break it into a multistage build so we don't have to worry about package cleanup. apt doesn't exist on alpine, the package manager is apk.

If you clone the repository in the windows file system, even though you run docker with WSL, it still throws an error where Dockerfile executes the bash script. Therefore, I have introduced dos2unix to replace CRLF with LF while building the image. However, I believe once we build the docker images and publish them to the registries, we should not have such a problem anyway. What do you think? @ksurl

**Comment by ksurl** [comment]
oh yeah I hadn't thought about that scenario. if you do use a prebuilt image then the file should already be LF. I think since it's a local environment issue (windows git has a setting to set the behavior globally or local to the repo on how to handle line endings), maybe just put a disclaimer or faq in the documentation. the most userfriendly option for windows to convert line endings would be a gui editor like notepad++. 2 clicks in the bottrom right corner
image

**Comment by barinali** [comment]
While we could quickly introduce a guide for that, I think it's relatively easy to overlook. I think we will publish a docker image soon and get rid of dos2unix.

We had a user in the discord server* needing some help about running docker compose in Windows. That session urged me to introduce dos2unix. By the way, feel free to join us there, as help is especially needed for installation nowadays. 💪🏻

**Comment by barinali** [comment]
@ksurl could you please prune your docker system (not like everything, but at least automatisch relevant layers/volumes/images) and try this? I get an error indicating `openssl` cannot be found.

**Comment by ksurl** [comment]
yeah i think alpine uses musl for c library and libressl. it may be an issue switching from ubuntu

**Comment by barinali** [comment]
While I doubt we need ca-certificates, wouldn't it be easier to have a pre-stage with ubuntu (or another relatively lean distro) and tackle openssl + dos2unix there? Then we could copy them over to the final stage. I think this way; it'll be less overhead.

**Comment by ksurl** [comment]
I'm not sure what is needed to copy over for those packages.

**Comment by barinali** [comment]
> I'm not sure what is needed to copy over for those packages.

Ah, you are certainly right. I was thinking on top of a previous iteration of the Dockerfile, where we got `openssl` within the Dockerfile, rather than in an entrypoint script. Nevermind me. We may try your suggestion.

**Comment by ksurl** [comment]
which service is having the openssl error? I am not seeing anything.

**Comment by barinali** [comment]
> which service is having the openssl error? I am not seeing anything.

After deleting the volume, it throws an error while using `openssl` without explicitly defining corresponding environment variables.

**Comment by ksurl** [comment]
I forgot I already added the package before rebuilding and running it again. I'll push the latest change.

**Review by barinali — APPROVED** [review]
@ksurl, thanks once again for your contribution!

**Pull request merged**